### DISCUSSION TITLE
Some errata for C++23 Best Practices, published 2024-02-16

### DISCUSSION THREAD

**Discussion by Chris1LDN** [lefticus/cpp_weekly#368] [C++ Best Practices Book]
Hi,

As discussed on twitter, I've been reading the C++23 Best Practices book, which I'm enjoying, but have found some issues I thought I'd share.

Page 29 - _"Sanitizers can only instrument code not removed by the optimizer, but the optimizer exploits some times of UB"_ - possibly meant to say "some types of UB"?

Page 40 - _"does not have the -fhardend flag yet"_ - should be "-fhardened"

Page 63 - _"Exercise: Research strong typedef libraries for C++"_ - just a comment... not sure if the word "typedef" is best to use here since modern C++ has replaced it with "using" and newcomers might not know what "typedef" means. Maybe "strong type libraries" or "strong type system libraries"? 

I'm only about half-way through but might not have time to read further for a while so thought I'd share what I've spotted so far.

**Comment by LB--** [comment]
Re: "strong typedef": the whole point is to get different behavior from `using` type aliases, so I think the name is fine. We want to **def**ine a new **type**, which just has all the same properties as an existing type and isn't implicitly convertible to the type it's based on. Basically what a beginner might expect the `typedef` keyword to do before learning what it actually does and being disappointed.
